src/libvirtApi/common.js

```diff
@@ -161,88 +165,103 @@ function doUsagePolling(name, connectionName, objPath) {
+            Object.assign(props, {
+                disksStats: calculateDiskStats(info)
+            });
+
+            logDebug(`doUsagePolling: ${JSON.stringify(props)}`);
+            store.dispatch(updateVm(props));
+        }
+    } catch (ex) {
+        console.warn(`GetStats(${name}, ${connectionName}) failed: ${ex.toString()}`);
+    }
+    delayPolling(() => doUsagePolling(name, connectionName, objPath), null);
 }
 
-function getLoggedInUser() {
-    logDebug(`GET_LOGGED_IN_USER:`);
-    return cockpit.user().then(loggedUser => store.dispatch(setLoggedInUser({ loggedUser })));
+async function getLoggedInUser() {
+    const loggedUser = await cockpit.user();
+    logDebug(`GET_LOGGED_IN_USER:`, loggedUser);
+    store.dispatch(setLoggedInUser({ loggedUser }));
 }
 
-export function getLibvirtVersion({ connectionName }) {
-    return call(connectionName, "/org/libvirt/QEMU", "org.freedesktop.DBus.Properties", "Get", ["org.libvirt.Connect", "LibVersion"], { timeout, type: "ss" })
-            .then(version => store.dispatch(updateLibvirtVersion({ libvirtVersion: version[0].v })));
+export async function getLibvirtVersion({ connectionName }) {
+    const [version] = await call(connectionName, "/org/libvirt/QEMU", "org.freedesktop.DBus.Properties", "Get", ["org.libvirt.Connect", "LibVersion"], { timeout, type: "ss" });
+    store.dispatch(updateLibvirtVersion({ libvirtVersion: version.v }));
 }
 
-function getNodeMaxMemory({ connectionName }) {
-    // Some nodes don"t return all memory in just one cell.
+async function getCapabilities({ connectionName }) {
+    try {
+        const [capabilitiesXML] = await call(connectionName, "/org/libvirt/QEMU", "org.libvirt.Connect", "GetCapabilities", [],
+                                             { timeout, type: "" });
+        const capabilities = parseDumpxmlForCapabilities(capabilitiesXML);
+        store.dispatch(setCapabilities({ capabilities }));
+    } catch (ex) {
+        console.warn("NodeGetMemoryStats failed:", ex.toString());
+        throw ex;
+    }
+}
+
+async function getNodeMaxMemory({ connectionName }) {
+    // Some nodes don't return all memory in just one cell.
     // Using -1 == VIR_NODE_MEMORY_STATS_ALL_CELLS will return memory across all cells
-    return call(connectionName, "/org/libvirt/QEMU", "org.libvirt.Connect", "NodeGetMemoryStats", [-1, 0], { timeout, type: "iu" })
-            .then(stats => store.dispatch(setNodeMaxMemory({ memory: stats[0].total })))
-            .catch(ex => {
-                console.warn("NodeGetMemoryStats failed:", ex.toString());
-                return Promise.reject(ex);
-            });
+    try {
+        const [stats] = await call(connectionName, "/org/libvirt/QEMU", "org.libvirt.Connect", "NodeGetMemoryStats", [-1, 0], { timeout, type: "iu" });
+        store.dispatch(setNodeMaxMemory({ memory: stats.total }));
+    } catch (ex) {
+        console.warn("NodeGetMemoryStats failed:", ex.toString());
+        throw ex;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1280-20240125-153407-61755b5b-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)